pkgs/by-name/wl/wlcs/package.nix

```diff
@@ -38,6 +48,9 @@ stdenv.mkDerivation (finalAttrs: {
     wayland-scanner # needed by cmake
   ];
 
+  # Whatever default optimisation level is in use here (can reproduce with -O3) breaks std::function usage in some tests
+  env.NIX_CFLAGS_COMPILE = "-O2";
```

Comment:
seems like `-Werror` is enabled.
```suggestion
  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-Wno-error=maybe-uninitialized";
```

I was able to get it to compile with out disabling the warning but it is weird.
<details><summary>5x.diff</summary>

```diff
diff --git a/tests/xdg_popup.cpp b/tests/xdg_popup.cpp
index 99c2a0b..41ee16d 100644
--- a/tests/xdg_popup.cpp
+++ b/tests/xdg_popup.cpp
@@ -734,6 +734,10 @@ INSTANTIATE_TEST_SUITE_P(
     Default,
     XdgPopupPositionerTest,
     testing::Values(
+        PositionerTestParams{"default values", (window_width - popup_width) / 2, (window_height - popup_height) / 2, PositionerParams()},
+        PositionerTestParams{"default values", (window_width - popup_width) / 2, (window_height - popup_height) / 2, PositionerParams()},
+        PositionerTestParams{"default values", (window_width - popup_width) / 2, (window_height - popup_height) / 2, PositionerParams()},
+        PositionerTestParams{"default values", (window_width - popup_width) / 2, (window_height - popup_height) / 2, PositionerParams()},
         PositionerTestParams{"default values", (window_width - popup_width) / 2, (window_height - popup_height) / 2, PositionerParams()}
     ));
 

```
</details>

Reply:
*Might* be this upstream issue? The explained situation there seems to be around LTO, but the error points at the same line…

https://github.com/canonical/wlcs/issues/346

Considering that the "fix" there was to just disable LTO in the Debian build code, and there are references to this being a compiler/linker issue, let's go with disabling the warning for now until upstream (whichever one that may be here) sorts that out.